client/my-sites/themes/atomic-transfer-dialog.tsx

```diff
@@ -0,0 +1,160 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { Dialog } from '@automattic/components';
+import { localize, translate } from 'i18n-calypso';
+import { Component } from 'react';
+import { connect } from 'react-redux';
+import EligibilityWarnings from 'calypso/blocks/eligibility-warnings';
+import Notice from 'calypso/components/notice';
+import { getSiteSlug } from 'calypso/state/sites/selectors';
+import {
+	acceptAtomicTransferDialog,
+	dismissAtomicTransferDialog,
+	activate as activateTheme,
+	initiateThemeTransfer,
+} from 'calypso/state/themes/actions';
+import {
+	getCanonicalTheme,
+	getThemeForAtomicTransferDialog,
+	isExternallyManagedTheme,
+	shouldShowAtomicTransferDialog,
+} from 'calypso/state/themes/selectors';
+import {
+	isTransferComplete,
+	isUploadInProgress,
+} from 'calypso/state/themes/upload-theme/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { Theme } from 'calypso/types';
+import './atomic-transfer-dialog.scss';
+
+interface AtomicTransferDialogProps {
+	siteId?: number;
+	inProgress?: boolean;
+	isTransferred?: boolean;
+	showEligibility: boolean;
+	theme: Theme;
+	siteSlug?: string | null;
+	isMarketplaceProduct?: boolean;
+	dispatchAcceptAtomicTransferDialog: typeof acceptAtomicTransferDialog;
+	dispatchDismissAtomicTransferDialog: typeof dismissAtomicTransferDialog;
+	dispatchActivateTheme: typeof activateTheme;
+	dispatchRecordTracksEvent: typeof recordTracksEvent;
+	dispatchInitiateThemeTransfer: typeof initiateThemeTransfer;
+}
+
+class AtomicTransferDialog extends Component< AtomicTransferDialogProps > {
+	handleAccept() {
+		const { siteId, dispatchInitiateThemeTransfer } = this.props;
+		if ( ! siteId ) {
+			return;
+		}
+
+		dispatchInitiateThemeTransfer( siteId, null, '' );
+	}
+
+	getCurrentPath = () => {
```

Comment:
Minor thought: we're only using this in one place, so we could pull the `.wordpress.com` => `.wpcomstaging.com` replacement into the function. If we do that, we could rename the function to `getAtomicSitePath()` or something similar.